src/Json/Decode/Applicative.elm

```diff
@@ -0,0 +1,19 @@
+module Json.Decode.Applicative exposing (apply, with)
+
+{-| Json.Decode.Applicative module
+
+This module implements typeclass applicative (apply : m (a -> b) -> a -> b)
+for Decoder functor based on the monadic return (succeed) and bind (andThen)
+
+-}
+
+import Json.Decode exposing (Decoder, andThen, succeed)
+
+
+apply : Decoder (a -> b) -> Decoder a -> Decoder b
+apply df da =
+    andThen (\a -> andThen (\f -> succeed (f a)) df) da
+
+
+with da df =
+    apply df da
```

Comment:
I can see the order of args in apply is probably more common across languages but in elm, I would probably define it like this `flip apply` version right away.